binary_port/src/error_code.rs

```diff
@@ -27,27 +27,165 @@ pub enum ErrorCode {
+    InvalidTransactionExcessiveSize = 35,
+    /// The transaction sent to the network had an excessive time to live
+    #[error("The transaction sent to the network had an excessive time to live")]
+    InvalidTransactionExcessiveTimeToLive = 36,
+    /// The transaction sent to the network had a timestamp located in the future.
+    #[error("The transaction sent to the network had a timestamp that has not yet occurred")]
+    InvalidTransactionTimestampInFuture = 37,
+    /// The transaction sent to the network had a provided body hash that conflicted with hash
+    /// derived by the network
+    #[error("The transaction sent to the network had an invalid body hash")]
+    InvalidTransactionBodyHash = 38,
+    /// The transaction sent to the network had a provided hash that conflicted with the hash
+    /// dreived by the network
+    #[error("The transaction sent to the network had an invalid body hash")]
+    InvalidTransactionHash = 39,
+    /// The transaction sent to the network had an emtpy approvals set
+    #[error("The transaction sent to the network had no approvals")]
+    InvalidTransactionEmptyApprovals = 40,
+    /// The transaction sent to the network had an invalid approval
+    #[error("The transaction sent to the network had an invalid approval")]
+    InvalidTransactionInvalidApproval = 41,
+    /// The transaction sent to the network had excessive args length
+    #[error("The transaction sent to the network had excessive args length")]
+    InvalidTransactionExcessiveArgsLength = 42,
+    /// The transaction sent to the network had excessive approvals
+    #[error("The transaction sent to the network had excessive approvals")]
+    InvalidTransactionExcessiveApprovals = 43,
+    /// The transaction sent to the network exceeds the block gas limit
+    #[error("The transaction sent to the network exceeds the networks block gas limit")]
+    InvalidTransactionExceedsBlockGasLimit = 44,
+    /// The transaction sent to the network had a missing arg
+    #[error("The transaction sent to the network was missing an argument")]
+    InvalidTransactionMissingArg = 45,
+    /// The transaction sent to the network had an argument with an unexpected type
+    #[error("The transaction sent to the network had an unexpected argument type")]
+    InvalidTransactionUnexpectedArgType = 46,
+    /// The transaction sent to the network had an invalid argument
+    #[error("The transaction sent to the network had an invalid argument")]
+    InvalidTransactionInvalidArg = 47,
+    /// The transaction sent to the network had an insufficient transfer amount
+    #[error("The transaction sent to the network had an insufficient transfer amount")]
+    InvalidTransactionInsufficientTransferAmount = 48,
+    /// The transaction sent to the network had a custom entry point when it should have a non
+    /// custom entry point.
+    #[error("The transaction sent to the network should not have a custom entry point")]
```

Comment:
```suggestion
    #[error("The native transaction sent to the network should not have a custom entry point")]
```

"custom" is allowed in non-native transactions